### PR TITLE
#27: Fixed SONAR code smells.

### LINKED ISSUE #27: Fix code smells in 1.2.0
## Situation

The project collected a number of code smells that need to be addressed in the version up to 1.2.0

## Acceptance Criteria

1. All code smells fixed.

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #27.

### CODE REVIEW

File: doc/changes/changes_1.2.1.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# JUnit5 System Extensions 1.2.1, released 2021-05-02
+
+Code name: Mute output
```
Comment (kaklakariada) [inline doc/changes/changes_1.2.1.md]: Same code name as before? ;)

File: src/main/java/org/itsallcode/junit/sysextensions/SystemErrGuard.java [hunk 2]
```diff
@@ -51,6 +51,7 @@ protected Namespace getNamespace()
         return NAMESPACE;
     }
 
+    @SuppressWarnings("java:S106")
```
Comment (kaklakariada) [inline src/main/java/org/itsallcode/junit/sysextensions/SystemErrGuard.java]: Please add a short comment with a justification for the exclusion. Not everyone knows what `java:S106` means ;)
```suggestion
    @SuppressWarnings("java:S106") // System.out used intentionally
```

File: src/main/java/org/itsallcode/junit/sysextensions/SystemOutGuard.java [hunk 3]
```diff
@@ -51,6 +51,7 @@ protected Namespace getNamespace()
         return NAMESPACE;
     }
 
+    @SuppressWarnings("java:S106")
```
Comment (kaklakariada) [inline src/main/java/org/itsallcode/junit/sysextensions/SystemOutGuard.java]: Please add a short justification
```suggestion
    @SuppressWarnings("java:S106") // System.err used intentionally
```

File: src/test/java/org/itsallcode/junit/sysextensions/TestSytemOutWithBuffer.java [hunk 4]
```diff
@@ -38,10 +38,12 @@ void beforeEach()
     {
     }
 
+    @SuppressWarnings("squid:S2699")
     @Test
     void testGetCapturedRegular(final Capturable stream) throws IOException
     {
-        // This test is necessary to see forEach behavior
+        // This test is necessary to see forEach behavior.
+        // It is intentionally empty.
```
Comment (kaklakariada) [inline src/test/java/org/itsallcode/junit/sysextensions/TestSytemOutWithBuffer.java:46]: This explains the suppressed warning 👍

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: doc/changes/changes_1.2.1.md (hunk 1 above)
Comment (redcatbear) [inline doc/changes/changes_1.2.1.md]:
```suggestion
Code name: Sonar smell fixes for 1.2.0
```

Comment (redcatbear) [reply doc/changes/changes_1.2.1.md]: Fixed.

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**